zokrates_core/src/compile.rs

```diff
@@ -129,20 +129,25 @@ impl fmt::Display for CompileErrorInner {
     }
 }
 
-// See zokrates_fs_resolver for the spec
-pub type Resolve<'a, E> = &'a dyn Fn(PathBuf, PathBuf) -> Result<(String, PathBuf), E>;
-
 type FilePath = PathBuf;
 
+pub trait Resolver<E: Into<imports::Error>> {
```

Comment:
Can `<E: Into<imports::Error>>` go on `resolve` rather than on the trait? I think it's good practice and it might also make things clearer